web/src/utils/factory-management/products.ts

```diff
@@ -10,16 +10,8 @@ export const addProductToFactory = (
     recipe?: string,
   }
 ) => {
-  // In order to allow multiple products of the same product but using different recipes to be added, we need to generate a unique ID using the product's ID as well as the recipe ID.
-
-  let productId = ''
-
-  if (options.id && options.recipe) {
```

Comment:
This was never the case as none of the function calls passed a `recipe`.  
I removed it as it broke my implementation otherwise

Reply:
Yeah this was the primary reason why the button was borked in the first place, I changed the expected data structure.